### PR TITLE
Added .pre-commit-config.yaml, updated requirements.txt

### PR DESCRIPTION
**Pull request by BaptisteB-A**
Added the following pre-commit hooks to Artemis :
- black
- trailing-whitespace
- check-ast
- check-merge-conflict
- flake8
- check-json
- check-yaml
- end-of-file-fixer
- pretty-format-json

We are waiting on the following PR to get it merged:
* [x] #326 
* [ ] #327 
* [ ] #331 
* [x] #336  
- [ ] https://github.com/CanalTP/artemis/pull/338

### CODE REVIEW

**Comment by mbounabi** [comment]
That's a dangerous game to run black on all files at once... And one dangerous commit for us to validate. Maybe we should discuss the strategy on how to introduce the pre-commit checks in Artemis

File: artemis/base_pytest.py [hunk 1]
```diff
@@ -16,33 +16,32 @@
 from artemis.configuration_manager import config
 from artemis.common_fixture import CommonTestFixture
 
-if six.PY3: # case using python 3
+if six.PY3:  # case using python 3
     from enum import Enum
-
-if six.PY2: # case using python 2
+elif six.PY2:  # case using python 2
```
Comment (AurelienLP) [inline artemis/base_pytest.py]: Really the same ?

File: artemis/docker_orchestrator.py [hunk 2]
```diff
@@ -131,36 +142,60 @@ def launch_coverages(coverages):
...
+                        instances=instance_render,
+                        kraken_env=kraken_env_parameters,
+                        jormun_env=jormun_env_parameters,
+                    )
+                )
 
             # Create and start containers
             kraken_name = "kraken-" + instance_name
-            upInstanceCommand = "TAG=dev docker-compose -f docker-compose.yml -f kirin/docker-compose_kirin.yml -f asgard/docker-compose_asgard.yml -f " + instance_file + " up -d --remove-orphans " + kraken_name + " instances_configurator"
+            upInstanceCommand = "TAG=dev docker-compose -f docker-compose.yml -f kirin/docker-compose_kirin.yml -f asgard/docker-compose_asgard.yml -f {} up -d --remove-orphans {} instances_configurator".format(
```
Comment (AurelienLP) [inline artemis/docker_orchestrator.py]: :heart_eyes:

File: artemis/tests/__init__.py [hunk 3]
```diff
@@ -1 +0,0 @@
-
```
Comment (AurelienLP) [inline artemis/tests/__init__.py]: Is this file needed ?

File: artemis/utils.py [hunk 4]
```diff
@@ -63,15 +63,19 @@ def get_ref(call_id):
     It would thus be possible to execute the tests on a dev computer and access the ref
     on the CI platform
     """
-    assert os.path.exists(config['REFERENCE_FILE_PATH']), \
-        "no reference directory found: {} does not exists".format(config['REFERENCE_FILE_PATH'])
+    assert os.path.exists(
```
Comment (AurelienLP) [inline artemis/utils.py]: Need to check if it still works

File: artemis/tests/__init__.py (hunk 3 above)
Comment (mbounabi) [reply artemis/tests/__init__.py]: yes, it is needed to import files from artemis.tests

**Comment by AurelienLP** [comment]
I've checked all files. I will approve after this https://github.com/CanalTP/artemis/pull/329#discussion_r330917664 an this https://github.com/CanalTP/artemis/pull/329#discussion_r330924572 would have been answered.

File: artemis/base_pytest.py (hunk 1 above)
Comment (AurelienLP) [reply artemis/base_pytest.py]: Yes, we use Python2 or 3 anyway...

File: artemis/utils.py (hunk 4 above)
Comment (BaptisteB-A) [reply artemis/utils.py]: Why wouldn't it work anymore ? Aren't the changes equivalent to what it was before ?

Comment (AurelienLP) [reply artemis/utils.py]: I don't know the syntax for assert... I expect it to be equivalent, but we are never to sure :smile:

File: artemis/docker_orchestrator.py [hunk 5]
```diff
@@ -164,12 +164,8 @@ def launch_coverages(coverages):
 
             # Create and start containers
             kraken_name = "kraken-" + instance_name
-            upInstanceCommand = (
-                "TAG=dev docker-compose -f docker-compose.yml -f kirin/docker-compose_kirin.yml -f asgard/docker-compose_asgard.yml -f "
-                + instance_file
-                + " up -d --remove-orphans "
-                + kraken_name
-                + " instances_configurator"
+            upInstanceCommand = "TAG=dev docker-compose -f docker-compose.yml -f kirin/docker-compose_kirin.yml -f asgard/docker-compose_asgard.yml -f {} up -d --remove-orphans {} instances_configurator".format(
```
Comment (BaptisteB-A) [inline artemis/docker_orchestrator.py]: Replaced pluses by a format() because of Flake8 returning an error

File: artemis/docker_orchestrator.py [hunk 6]
```diff
@@ -190,21 +186,15 @@ def launch_coverages(coverages):
 
             # Delete instance container
             # The command is divided in 2 separate commands to be handled on old versions of docker/docker-compose
-            stopCommand1 = (
-                "TAG=dev docker-compose -f docker-compose.yml -f kirin/docker-compose_kirin.yml -f asgard/docker-compose_asgard.yml -f "
-                + instance_file
-                + " stop "
-                + kraken_name
+            stopCommand1 = "TAG=dev docker-compose -f docker-compose.yml -f kirin/docker-compose_kirin.yml -f asgard/docker-compose_asgard.yml -f {} stop {}".format(
+                instance_file, kraken_name
```
Comment (BaptisteB-A) [inline artemis/docker_orchestrator.py]: Replaced pluses by a format() because of Flake8 returning an error

File: artemis/docker_orchestrator.py [hunk 7]
```diff
@@ -190,21 +186,15 @@ def launch_coverages(coverages):
...
             print("Run : {}".format(stopCommand1))
             wait_for_kraken_stop(kraken_name)
 
-            stopCommand2 = (
-                "TAG=dev docker-compose -f docker-compose.yml -f kirin/docker-compose_kirin.yml -f asgard/docker-compose_asgard.yml -f "
-                + instance_file
-                + " rm -f "
-                + kraken_name
+            stopCommand2 = "TAG=dev docker-compose -f docker-compose.yml -f kirin/docker-compose_kirin.yml -f asgard/docker-compose_asgard.yml -f {} rm -f {}".format(
+                instance_file, kraken_name
```
Comment (BaptisteB-A) [inline artemis/docker_orchestrator.py]: Replaced pluses by a format() because of Flake8 returning an error

File: .pre-commit-config.yaml [hunk 8]
```diff
@@ -0,0 +1,16 @@
+repos:
+-   repo: https://github.com/ambv/black
+    rev: 18.9b0
+    hooks:
+    - id: black
+      language_version: python3.6
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v2.3.0  # Use the ref you want to point at
+    hooks:
```
Comment (nonifier) [inline .pre-commit-config.yaml]: Two other hooks could also be interesting:
- end-of-file-fixer
- pretty-format-json

**Comment by kinnou02** [comment]
This will have to be squash :)

File: .pre-commit-config.yaml (hunk 8 above)
Comment (BaptisteB-A) [reply .pre-commit-config.yaml]: Done :+1:

File: artemis/readme.md [hunk 9]
```diff
@@ -7,20 +7,20 @@ aka ArtemisNG has been designed to run tests using docker-compose, in order to b
 ### 1. Run Navitia services in docker containers
 
 *   You can clone the repository navitia-docker-compose, available [here](https://github.com/CanalTP/navitia-docker-compose) and follow the instructions in the [readme](https://github.com/CanalTP/navitia-docker-compose/blob/master/README.md) file to have the whole Navitia environment running.
-    - Basic usage: launch this command at the root directory of the repo 'navitia-docker-compose':
```
[the remark this answers is not in the document]
Comment (BaptisteB-A) [reply artemis/readme.md]: Thanks, it should be fine now with [https://github.com/CanalTP/artemis/pull/329/commits/14bef23137c6f7b4a7fd2877d4df4b2152b35943](url)

[the remark this answers is not in the document]
Comment (pbougue) [reply artemis/readme.md]: Nice, it's a way better syntax! This needs to be adopted on every repo (and slate doc in navitia if it works there) :tada: 
You forgot 2 lines though :wink:

[the remark this answers is not in the document]
Comment (BaptisteB-A) [reply artemis/readme.md]: Oops! I added the backslashes for these 2 lines

File: .pre-commit-config.yaml [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+    -   id: trailing-whitespace
+    -   id: check-ast
+    -   id: check-merge-conflict
+    -   id: check-json
+    -   id: check-yaml
+        language_version: python2.7
+    -   id: flake8
```
[the remark this answers is not in the document]
Comment (BaptisteB-A) [reply .pre-commit-config.yaml]: Done [here](https://github.com/CanalTP/artemis/pull/329/commits/343efee212aaa214f2ac65ee14e6371876fa5b6c)

[the remark this answers is not in the document]
Comment (BaptisteB-A) [reply .pre-commit-config.yaml]: Done [here](https://github.com/CanalTP/artemis/pull/329/commits/343efee212aaa214f2ac65ee14e6371876fa5b6c)

**Comment by pbougue** [comment]
To avoid any brutal merge on that, I added `do_not_merge` tag :
* This PR will probably conflict with all the ongoing ones and is probably less urgent.
* For safety it's probably better to apply conf from scratch once there is no big Artemis PR in the pipe

But the work is OK for me, and I won't block it :+1:
